rcl/test/CMakeLists.txt

```diff
@@ -40,78 +37,112 @@ if(NOT WIN32)
     if(NOT WIN32)
       set_target_properties(test_memory_tools PROPERTIES COMPILE_FLAGS "-std=c++11")
     endif()
-    target_link_libraries(test_memory_tools ${PROJECT_NAME} ${extra_test_libraries})
+    target_link_libraries(test_memory_tools ${extra_test_libraries})
   endif()
 endif()
 
-ament_add_gtest(test_allocator rcl/test_allocator.cpp
-  ENV ${extra_memory_tools_env} APPEND_LIBRARY_DIRS ${rcl_lib_dir})
-if(TARGET test_allocator)
-  target_include_directories(test_allocator PUBLIC
-    ${rcl_interfaces_INCLUDE_DIRS}
-    ${rmw_INCLUDE_DIRS}
-  )
-  if(NOT WIN32)
-    set_target_properties(test_allocator PROPERTIES COMPILE_FLAGS "-std=c++11")
+macro(test_target)
+  find_package(${rmw_implementation} REQUIRED)
+  ament_add_gtest(test_allocator${target_suffix} rcl/test_allocator.cpp
+    ENV ${extra_memory_tools_env} APPEND_LIBRARY_DIRS ${rcl_lib_dir})
+  if(TARGET test_allocator${target_suffix})
+    target_include_directories(test_allocator${target_suffix} PUBLIC
```

Comment:
so I commented this out in my commit for the publisher test only, but I don't think target_include_directories is needed here.